nmstatectl/nmstatectl.py

```diff
@@ -98,6 +125,77 @@ def _filter_state(state, whitelist):
     return state
 
 
+def edit(args):
+    state = _filter_state(netinfo.show(), args.only)
+
+    if args.yaml:
+        suffix = '.yaml'
+        txtstate = pretty_yaml(state)
+    else:
+        suffix = '.json'
+        txtstate = pretty_json(state)
+    with tempfile.NamedTemporaryFile(suffix=suffix,
+                                     prefix='nmstate-') as statefile:
+        statefile.write(txtstate)
+        statefile.flush()
+
+        while True:
+            txtstate = run_editor(txtstate, suffix)
+            error = ""
+
+            if txtstate is None:
+                return 1
+
+            try:
+                if args.yaml:
+                    new_state = yaml.load(txtstate)
+                else:
+                    new_state = json.loads(txtstate)
+            except ValueError as e:
+                error = "Invalid JSON syntax: %s\n" % e
+            except yaml.parser.ParserError as e:
+                error = "Invalid YAML syntax: %s\n" % e
+
+            if not error and (not new_state or 'interfaces' not in new_state):
+                error = "Invalid state: should contain 'interfaces' entry.\n"
+
+            if error:
+                sys.stderr.write("ERROR: ")
+                sys.stderr.write(error)
+                response = ""
+                while response not in ("y", "n"):
+                    response = input("Try again? [y,n]:\n"
```

Comment:
It will be nice to accept both upper and lower cases input. (normalize to lower-case).

Reply:
changed